src/operator/regression_output-inl.h

```diff
@@ -95,12 +125,35 @@ void RegressionForward(const nnvm::NodeAttrs& attrs,
   });
 }
 
+template<typename xpu, typename ForwardOp>
+void RegressionForwardEx(const nnvm::NodeAttrs& attrs,
+                         const OpContext& ctx,
+                         const std::vector<NDArray>& inputs,
+                         const std::vector<OpReqType>& req,
+                         const std::vector<NDArray>& outputs) {
+  CHECK_EQ(inputs.size(), 2U);
```

Comment:
I suggest we always explicitly check stypes in ForwardEx/BackwardEx and call xxxImpl depending on storage types. For forward we only need to pass `inputs[0].data()` to `xxxImpl` since forward doesn't care about what the label is.

Reply:
still need to check stypes of the input data